### PR TITLE
[feature]:support spark connector sink data to doris

### PR DESCRIPTION
**Pull request by Kyofin**
## Proposed changes

support spark conncetor write dataframe to doris

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala [hunk 1]
```diff
@@ -17,14 +17,77 @@
...
+
+    val splitHost = beHostPort.split(";")
+    val choosedBeHost = splitHost(Random.nextInt(splitHost.length))
+    val dorisStreamLoader = new DorisStreamLoad(choosedBeHost, dbName, tbName, user, password)
+
+    data.foreachPartition(partition => {
+
+      val buffer = ListBuffer[String]()
+      partition.foreach(row => {
+        val rowString = row.toSeq.mkString("\t")
```
Comment (JNSimba) [inline extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala:65]: Do need to consider the **Null** value ？

File: extension/spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            StringBuilder sb = new StringBuilder();
+            sb.append("status: ").append(status);
+            sb.append(", resp msg: ").append(respMsg);
+            sb.append(", resp content: ").append(respContent);
+            return sb.toString();
+        }
+    }
+
+    public void load(String value) throws StreamLoadException {
+        LoadResponse loadResponse = loadBatch(value);
```
Comment (hf200012) [inline extension/spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java:118]: It is recommended to add the maximum number of failed retries ，Avoid failures caused by short-term network jitter

File: extension/spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                }
+            } catch (IOException e) {
+                throw new StreamLoadException(e);
+            }
+        }
+    }
+
+    private LoadResponse loadBatch(String value) {
+        Calendar calendar = Calendar.getInstance();
+        String label = String.format("audit_%s%02d%02d_%02d%02d%02d_%s",
```
Comment (hf200012) [inline extension/spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java:137]: It is recommended to use spark_connector_ for the label name, which is easy to distinguish

File: extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisOptions.scala [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.apache.doris.spark.sql
+
+object DorisOptions {
+  val beHostPort="beHostPort"
```
Comment (hf200012) [inline extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisOptions.scala:20]: The IP address of FE should be configured here, and the list of surviving BE nodes can be obtained through FE, and then communication training or other strategies during Stream load, connect to BE to execute load, and avoid the pressure of FE

**Review by hf200012 — COMMENTED** [review]
1. It is recommended to add the number of failed load retries
2. It is recommended to get the list of surviving BE nodes through FE, and directly connect to BE to execute Stream load through rotation training or other strategies.
3. Add to execute stream load according to the time interval, two strategies, one is according to the number of records you have implemented so far, the other is according to the time interval

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**